### PR TITLE
Abstraction allow using different file formats

### PR DESCRIPTION
**Pull request by Warxcell**
As PHP, xml and others. 

There is a more work todo. Let's discuss it. :)

### CODE REVIEW

File: PhpProcessor.php [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+{
+
+    public function dump($file, $values)
+    {
+        
+    }
+
+    public function parse($file)
+    {
+
```
Comment (OskarStark) [inline PhpProcessor.php]: No blank lines please

File: Processor.php [hunk 2]
```diff
@@ -4,16 +4,38 @@
...
 use Symfony\Component\Yaml\Inline;
-use Symfony\Component\Yaml\Parser;
-use Symfony\Component\Yaml\Yaml;
 
 class Processor
 {
     private $io;
 
+    /**
+     * @var <FileProcessorInterface>
```
Comment (OskarStark) [inline Processor.php]: Why < andere > ???

File: YamlProcessor.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+namespace Incenteev\ParameterHandler;
+
+use Symfony\Component\Yaml\Inline;
+use Symfony\Component\Yaml\Parser;
+use Symfony\Component\Yaml\Yaml;
+
+class YamlProcessor implements FileProcessorInterface
+{
+    private $yamlParser;
```
Comment (OskarStark) [inline YamlProcessor.php]: Phpdoc missing

File: Processor.php (hunk 2 above)
Comment (Warxcell) [inline Processor.php]: Indicates that it's array of FileProcessorInterface.

File: PhpProcessor.php (hunk 1 above)
Comment (Warxcell) [inline PhpProcessor.php]: This is still todo. If you think it's ok that way I will continue it.

File: Processor.php (hunk 2 above)
Comment (OskarStark) [inline Processor.php]: didn't know this, i already use `FileProcessorInterface[]`

File: PhpProcessor.php (hunk 1 above)
Comment (OskarStark) [inline PhpProcessor.php]: 👍

File: Processor.php (hunk 2 above)
Comment (Warxcell) [inline Processor.php]: NetBeans support it. I guess it's not a problem to change it to: `FileProcessorInterface[]`

Comment (Ninir) [inline Processor.php]: I would go for `FileProcessorInterface[]` as it is best known

**Pull request closed without merging**